crates/light-client-prover/src/circuit/initial_values.rs

```diff
@@ -0,0 +1,344 @@
+        match const_hex::const_decode_to_array(
+            b"03796a3a8a86ff1cc37437585f0450f6059c397c01bce06bfbaaa36242f7ebfc02",
+        ) {
+            Ok(pub_key) => pub_key,
+            Err(_) => {
+                panic!("METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY must be valid 33-byte hex string")
+            }
+        };
+    pub const DEVNET_METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY: [u8; 33] =
+        match const_hex::const_decode_to_array(
+            b"0388e988066db18e19750fa92aa0fbf9c85104be2b5b507ce0aa7f30f3fe24b1ac",
+        ) {
+            Ok(pub_key) => pub_key,
+            Err(_) => {
+                panic!("METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY must be valid 33-byte hex string")
+            }
+        };
+
+    pub const NIGHTLY_METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY: [u8; 33] = {
+        let hex_pub_key = match option_env!("METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY") {
+            Some(hex_pub_key) => hex_pub_key,
+            None => "0313c4ff65eb94999e0ac41cfe21592baa52910f5a5ada9074b816de4f560189db",
+        };
+
+        match const_hex::const_decode_to_array(hex_pub_key.as_bytes()) {
+            Ok(pub_key) => pub_key,
+            Err(_) => {
+                panic!("METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY must be valid 33-byte hex string")
+            }
+        }
+    };
+
+    pub const TEST_NETWORK_WITH_FORKS_METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY: [u8; 33] = {
+        let hex_pub_key = match option_env!("METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY") {
+            Some(hex_pub_key) => hex_pub_key,
+            None => "0313c4ff65eb94999e0ac41cfe21592baa52910f5a5ada9074b816de4f560189db",
+        };
+
+        match const_hex::const_decode_to_array(hex_pub_key.as_bytes()) {
+            Ok(pub_key) => pub_key,
+            Err(_) => {
+                panic!("METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY must be valid 33-byte hex string")
+            }
+        }
+    };
+}
+
+#[cfg(feature = "native")]
+pub trait InitialValueProvider<Das: DaSpec> {
+    fn get_l2_genesis_root(&self) -> [u8; 32];
+
+    fn initial_batch_proof_method_ids(&self) -> Vec<(u64, [u32; 8])>;
+
+    fn batch_prover_da_public_key(&self) -> [u8; 33];
+
+    fn method_id_upgrade_authority_da_public_key(&self) -> [u8; 33];
+}
+
+#[cfg(feature = "native")]
+impl InitialValueProvider<MockDaSpec> for Network {
+    fn get_l2_genesis_root(&self) -> [u8; 32] {
+        match self {
+            Network::Nightly => mockda::GENESIS_ROOT,
+            _ => {
+                panic!("Only nightly allowed on mock da!")
+            }
+        }
+    }
+
+    fn initial_batch_proof_method_ids(&self) -> Vec<(u64, [u32; 8])> {
+        mockda::INITIAL_BATCH_PROOF_METHOD_IDS.to_vec()
+    }
+
+    fn batch_prover_da_public_key(&self) -> [u8; 33] {
+        mockda::BATCH_PROVER_DA_PUBLIC_KEY
+    }
+
+    fn method_id_upgrade_authority_da_public_key(&self) -> [u8; 33] {
+        mockda::METHOD_ID_UPGRADE_AUTHORITY_DA_PUBLIC_KEY
+    }
```

Comment:
make the nightly check on each of these methods. no need for a match, just go assert_eq(network, nightly)